pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -241,6 +251,16 @@ func (r *repository) Manifests(ctx context.Context, options ...distribution.Mani
 // Blobs returns a blob store which can delegate to remote repositories.
 func (r *repository) Blobs(ctx context.Context) distribution.BlobStore {
 	repo := repository(*r)
+
+	if r.pullthrough {
+		// Add to the context the BlobGetterService that provide access to remote servers.
+		// It will be used in pullthroughBlobStore.
```

Comment:
Mention that it needs to be instantiated here in order to share the cache for multiple stat calls made in descendant blob stores.